pages/blog/socketio-part1.md

```diff
@@ -0,0 +1,64 @@
+
+## Setting the scene
+
+Earlier this year my team undertook the task of re-implementing the backend of a realtime chat application – mentioning _re-implementing_ and _backend_ in the very first sentence is probably a PTSD trigger for many of you, but thankfully this post has nothing to do with rewriting-from-scratch horror stories. The app was originally built using a 3rd party push-notifications platform which allowed us to deliver a functional MVP in a relatively speedy manner. However, as new requirements started creeping in from the business, it was clear to the team that this 3rd party dependency was not really worth it anymore. We thus had to take the (arguably not so easy) decision to implement our own realtime API to gain complete control of each server connection. The design meeting would soon follow, to answer questions like “What framework should we use?” or “Would serverless make sense?”. The stack of the team, namely Python, [Flask](https://flask.palletsprojects.com/) and [Kubernetes](https://kubernetes.io/), pointed us in the direction of [Flask-SocketIO](https://flask-socketio.readthedocs.io/), making Socket.IO the protocol of choice. Although pragmatic, this stack driven design approach felt very unorthodox. The tooling dictated the choice of the client<->server communication protocol rather than the other way around. It was a decision primarily based on the stack and the expertise of the backend team. What if the Socket.IO client library for Swift is not maintained anymore? What if Socket.IO is a very verbose protocol, not friendly for clients with limited network connectivity? Thankfully, this is not the case and Socket.IO happens to be a well supported and carefully designed protocol. Nonetheless, the decision was made and the development of a Socket.IO API was about to commence.
+
+In the spirit of not repeating the sins of the past (i.e. not focusing on the client<->server interface), I started looking into how one can document a Socket.IO API. By document, I mean putting together some sort of spec or contract that can be agreed upon and shared with the consumers of the API. To my surprise, I couldn’t find a lot of resources on the topic. Coming from the REST world (where [OpenAPI](https://www.openapis.org/) is nowadays the de facto standard) and from the [GraphQL](https://graphql.org/) world (where everything is strictly typed), I found myself rather disappointed. It feels like the industry has been treating the documentation of event-driven APIs, such as Socket.IO or WebSocket ones, as a niche area which is not worth standardising. This is odd given how popular event-driven architectures have become over the past decade.
```

Comment:
```suggestion
In the spirit of not repeating the sins of the past (i.e. not focusing on the client<->server interface), I started looking into how one can document a Socket.IO API. By document, I mean putting together some sort of spec or contract that can be agreed upon and shared with the consumers of the API. To my surprise, I couldn’t find a lot of resources on the topic. Coming from the REST world (where [OpenAPI](https://www.openapis.org/) is nowadays the de facto standard) and from the [GraphQL](https://graphql.org/) world (where everything is strictly typed), I found myself rather disappointed. It felt like the industry had been treating the documentation of event-driven APIs, such as Socket.IO or WebSocket ones, as a niche area not worth standardizing. This felt odd, given how popular event-driven architectures had become over the past decade.
```

Reply:
So there was a typo + I changed the verbs to *past tense*. I did this because you start this paragraph in past tense and even the following paragraph is in the past tense as well. It read confusingly to jump around from past to present and past tense in this current blog post, so I decided to recommend you keep this part of your story in the past tense. 🙂